### PR TITLE
chore: add google analytics

### PR DESCRIPTION
**Pull request by subotic**
resolves DSP-1697

### CODE REVIEW

**Comment by tobiasschweizer** [comment]
Which legal implications does this have?

**Comment by subotic** [comment]
> Which legal implications does this have?

Legal implications?

**Comment by subotic** [comment]
> > > Which legal implications does this have?
> > 
> > 
> > Legal implications?
> 
> Google analytics tracks user info so I suppose a message has to be displayed whether the user agrees or disagrees.

Yes, you are right. We would need such a message. We would also need it without GA, because of cookies. We will have it at some point.

**Comment by tobiasschweizer** [comment]
> > > > Which legal implications does this have?
> > > 
> > > 
> > > Legal implications?
> > 
> > 
> > Google analytics tracks user info so I suppose a message has to be displayed whether the user agrees or disagrees.
> 
> Yes, you are right. We would need such a message. We would also need it without GA, because of cookies. We will have it at some point.

I am not sure about regulations but I guess we would need it right away.

**Comment by kilchenmann** [comment]
> > > > > Which legal implications does this have?
> > > > 
> > > > 
> > > > Legal implications?
> > > 
> > > 
> > > Google analytics tracks user info so I suppose a message has to be displayed whether the user agrees or disagrees.
> > 
> > 
> > Yes, you are right. We would need such a message. We would also need it without GA, because of cookies. We will have it at some point.
> 
> I am not sure about regulations but I guess we would need it right away.

We can do it the same way as we did in DSP-App... but at the moment it's not activated anymore

File: app/public/index.html [hunk 1]
```diff
@@ -76,6 +76,14 @@
...
     <script src="js/cytoscape-cola.js"></script>
     <script src="js/jquery.clipurl.js"></script>
+	<!-- Global site tag (gtag.js) - Google Analytics -->
+	<script async src="https://www.googletagmanager.com/gtag/js?id=G-CN60S2F3N4"></script>
+	<script>
+		window.dataLayer = window.dataLayer || [];
+		function gtag(){dataLayer.push(arguments);}
+		gtag('js', new Date());
+		gtag('config', 'G-CN60S2F3N4');
+	</script>
```
Comment (kilchenmann) [inline app/public/index.html:86]: Just a small, irrelevant detail: it looks like the indent is different than above.

**Review by kilchenmann — APPROVED** [review]

**Comment by subotic** [comment]
Regarding the user consent form, the way it was done in DSP-APP is enough to comply with the Swiss law. It is not enough to comply to the current EU law. For this we need a proper consent form which will be done next (see issues under DSP-1670).

**Comment by kilchenmann** [comment]
> Regarding the user consent form, the way it was done in DSP-APP is enough to comply with the Swiss law. It is not enough to comply to the current EU law. For this we need a proper consent form which will be done next (see issues under [DSP-1670](https://dasch.myjetbrains.com/youtrack/issue/DSP-1670)).

I know. It was anyway a quick and temporary solution. When I implemented it we didn't use cookies.

**Comment by subotic** [comment]
> > Regarding the user consent form, the way it was done in DSP-APP is enough to comply with the Swiss law. It is not enough to comply to the current EU law. For this we need a proper consent form which will be done next (see issues under [DSP-1670](https://dasch.myjetbrains.com/youtrack/issue/DSP-1670)).
> 
> I know. It was anyway a quick and temporary solution. When I implemented it we didn't use cookies.

I was not implying, that what you did was not good. At that time, it was everything we needed. GDPR came after and now we need a bit more. Either that or we need to block all traffic from the EU.

**Pull request merged**